dask/array/core.py

```diff
@@ -2125,7 +2148,10 @@ def from_array(x, chunks, name=None, lock=False, asarray=True, fancy=True,
                     dtype=x.dtype)
         dsk[original_name] = x
 
-    return Array(dsk, name, chunks, dtype=x.dtype)
+    nd = x.ndim if hasattr(x, 'ndim') else len(x.shape)
```

Comment:
So we could simplify this a bit by doing...

```
getattr(x, "ndim", len(x.shape))
```

Alternatively we could just always use `len(x.shape)` if it always works.

Just out of curiosity, which things did you encounter that lacked `.ndim`?

Reply:
> Just out of curiosity, which things did you encounter that lacked `.ndim`?

Another thing I can't reproduce anymore. It may have been a bug in my own implementation. Getting rid of it for now.